src/libraries/Microsoft.Extensions.Http/src/Logging/NetEventSource.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Extensions.Http
+{
+    [EventSource(Name = "Private.InternalDiagnostics.Microsoft.Extensions.Http")]
+    internal sealed class NetEventSource : EventSource
+    {
+        public static readonly NetEventSource Log = new NetEventSource();
+
+        public static class Keywords
+        {
+            public const EventKeywords Default = (EventKeywords)0x0001;
+            public const EventKeywords Debug = (EventKeywords)0x0002;
+        }
+
+        private const int CleanupCycleStartEventId = 1;
+        private const int CleanupCycleEndEventId = 2;
+        private const int CleanupItemFailedEventId = 3;
+        private const int HandlerExpiredEventId = 4;
+
+        [NonEvent]
+        public static void CleanupCycleStart(int initialCount)
+        {
+            Debug.Assert(Log.IsEnabled());
+            Log.CleanupCycleStart($"Starting HttpMessageHandler cleanup cycle with {initialCount} items");
+        }
+
+        [Event(CleanupCycleStartEventId, Keywords = Keywords.Debug, Level = EventLevel.Verbose)]
+        public void CleanupCycleStart(string message) =>
```

Comment:
Is this public on purpose? Should the `Debug.Assert` rather be in this method? I'm just comparing this with the shared `NetEventSource`.

Reply:
As I see, we have the `[Event]` ones both private or public randomly 😄 I guess it doesn't matter that much, since it's an internal class? But I think the `Debug.Assert` should be in `[NonEvent]` one
https://github.com/dotnet/runtime/blob/8ba85b4ff4226f2ef9fef94f4c99b20becbfd71d/src/libraries/System.Net.Http/src/System/Net/Http/NetEventSource.Http.cs#L55-L64